avocado/core/safeloader.py

```diff
@@ -103,20 +103,28 @@ def is_matching_klass(self, klass):
                     return True
         return False
 
+    @staticmethod
+    def _get_adjusted_path_for_level(statement, path):
+        level = getattr(statement, 'level', 0)
+        if level is None:
+            level = 0
```

Comment:
I could not spot in which case this `if` statement is needed. The previous line should be handling it. Here are the tests I did:

```
 >>> a = None
>>> getattr(a, 'level', 0)
0
>>> a = {'level': None}
>>> getattr(a, 'level', 0)
0
>>>
```

Does this cover other cases?

Reply:
> I could not spot in which case this `if` statement is needed. The previous line should be handling it. Here are the tests I did:
> 
> ```
>  >>> a = None
> >>> getattr(a, 'level', 0)
> 0
> >>> a = {'level': None}
> >>> getattr(a, 'level', 0)
> 0
> >>>
> ```
> 
> Does this cover other cases?

If the object actually has an attribute, and it's value is `None`, than this is needed.  Example:

```python
class A:
    level = None

class B:
    level = 0

class C:
    pass

print(getattr(A(), 'level', 0))
print(getattr(B(), 'level', 0))
print(getattr(C(), 'level', 0))
```

Results in:

```
None
0
0
```

The behavior in the `ast` module chanced from Python 3.8 to 3.9 (which will always have the attribute).  You can see the Fedora 32 (Python 3.8) job passing and the Fedora 33 (Python 3.9) job failing here:

https://cirrus-ci.com/build/6623712624508928

Reply:
Okay, got it; I was expecting the `statement` argument to be always a dictionary, or `None`.